TSOClient/tso.simantics/NetPlay/EODs/Handlers/VMEODSlotsPlugin.cs

```diff
@@ -0,0 +1,678 @@
+                        if (IsObjectBalanceInBounds() == true)
+                        {
+                            CalculateWheelStops(machineOdds);
+                            userClient.Send("slots_player_init", new byte[3] { Convert.ToByte(args[1]), Convert.ToByte(args[2]), 1 });
+                            State = VMEODSlotsStates.Idle;
+                        }
+                        else
+                        {
+                            userClient.Send("slots_player_init", new byte[3] { Convert.ToByte(args[1]), Convert.ToByte(args[2]), 0 });
+                            State = VMEODSlotsStates.Off;
+                        }
+                    }
+                    else
+                    {
+                        Console.WriteLine("There was an error trying to get this slot machine's balance.");
+                        machineBalance = -1;
+                        userClient.Send("slots_player_init", new byte[3] { Convert.ToByte(args[1]), Convert.ToByte(args[2]), 0 });
+                        State = VMEODSlotsStates.Off;
+                    }
+                });
+            }
+            else // user is owner
+            {
+                // I'm sure there's a more elegant way, but get the type of machine by the 2nd letter in its name
+                var name = Server.Object.Name;
+                byte machineType = 0;
+                if ((name != null) && (name.Length > 1)) {
+                    switch (name[1]) {
+                        // "Jack of Hearts" Slot Machine
+                        case 'J':
+                            {
+                                machineType = 2;
+                                break;
+                            }
+                        // "Gypsy Queen" Slot Machine
+                        case 'G':
+                            {
+                                machineType = 1;
+                                break;
+                            }
+                        // Viva PGT Home Casino - note lack of quote marks
+                        default:
+                            {
+                                machineType = 0;
+                                break;
+                            }
+                    }
+                }
+                State = VMEODSlotsStates.Managing;
+                int tempMachineBalance = (16384 * args[3]) + args[2];
+                userClient.Send("slots_owner_init", new byte[5] { Convert.ToByte(args[1]), Convert.ToByte(tempMachineBalance % 255),
+                    Convert.ToByte(tempMachineBalance / 255), Convert.ToByte(args[4]), machineType });
+            }
+        }
+        public override void OnDisconnection(VMEODClient client)
+        {
+            base.OnDisconnection(client);
+        }
+        private void ToggleOnOffHandler(string evt, string OnOffState, VMEODClient client)
+        {
+            onOffState = Int16.Parse(OnOffState);
```

Comment:
ALSO - Is the person using this actually the owner of the machine? As is, looks like anyone can turn the machine off.